src/common/status.h

```diff
@@ -376,3 +377,21 @@ struct [[nodiscard]] StatusOr {
     if (!status) return std::forward<decltype(status)>(status); \
     std::forward<decltype(status)>(status);                     \
   }).GetValue()
+
+#define RETURN_IF_ERROR(expr)                                        \
+  ({                                                                 \
+    auto&& status = (expr);                                          \
+    if (!status.ok()) return std::forward<decltype(status)>(status); \
+  })
+
+#define RETURN_IF_ERR_FROM_ROCKSDB(expr, error_code, format_str, with_status_string) \
+  ({                                                                                 \
+    auto&& status = (expr);                                                          \
+    if (!status.ok()) {                                                              \
+      if (with_status_string) {                                                      \
+        return {error_code, fmt::format(format_str, status.ToString())};             \
+      } else {                                                                       \
+        return {error_code, fmt::format(format_str)};                                \
+      }                                                                              \
+    }                                                                                \
+  })
```

Comment:
Several points:
- Many code of this macro can be implemented by a ctor `Status(rocksdb::Status&&)`, we don't need write them in the macro;
- After such ctor is impl, this macro is useless since it also can be handled by `GET_OR_RET`;
- This macro doesn't check the input type, also doesn't explicit specify the output type, which can trigger some weird issues (the return type can actually not a `Status`, but `rocksdb::Status`, or even `pair<int, string>`).